pyro/distributions/transforms/affine_autoregressive.py

```diff
@@ -220,6 +223,106 @@ def _inverse_stable(self, y):
+    >>> context_dim = 4
+    >>> batch_size = 3
+    >>> hidden_dims = [10*input_dim, 10*input_dim]
+    >>> base_dist = dist.Normal(torch.zeros(input_dim), torch.ones(input_dim))
+    >>> hypernet = ConditionalAutoRegressiveNN(input_dim, context_dim, hidden_dims)
+    >>> transform = ConditionalAffineAutoregressive(hypernet)
+    >>> pyro.module("my_transform", transform)  # doctest: +SKIP
+    >>> z = torch.rand(batch_size, context_dim)
+    >>> flow_dist = dist.ConditionalTransformedDistribution(base_dist,
+    ... [transform]).condition(z)
+    >>> flow_dist.sample(sample_shape=torch.Size([batch_size]))  # doctest: +SKIP
+
+    The inverse of the Bijector is required when, e.g., scoring the log density of a
+    sample with :class:`~pyro.distributions.TransformedDistribution`. This
+    implementation caches the inverse of the Bijector when its forward operation is
+    called, e.g., when sampling from
+    :class:`~pyro.distributions.TransformedDistribution`. However, if the cached
+    value isn't available, either because it was overwritten during sampling a new
+    value or an arbitary value is being scored, it will calculate it manually. Note
+    that this is an operation that scales as O(D) where D is the input dimension,
+    and so should be avoided for large dimensional uses. So in general, it is cheap
+    to sample from IAF and score a value that was sampled by IAF, but expensive to
+    score an arbitrary value.
+
+    :param autoregressive_nn: an autoregressive neural network whose forward call
+        returns a real-valued mean and logit-scale as a tuple
+    :type autoregressive_nn: nn.Module
+    :param log_scale_min_clip: The minimum value for clipping the log(scale) from
+        the autoregressive NN
+    :type log_scale_min_clip: float
+    :param log_scale_max_clip: The maximum value for clipping the log(scale) from
+        the autoregressive NN
+    :type log_scale_max_clip: float
+    :param sigmoid_bias: A term to add the logit of the input when using the stable
+        tranform.
+    :type sigmoid_bias: float
+    :param stable: When true, uses the alternative "stable" version of the transform
+        (see above).
+    :type stable: bool
+
+    References:
+
+    [1] Diederik P. Kingma, Tim Salimans, Rafal Jozefowicz, Xi Chen, Ilya Sutskever,
+    Max Welling. Improving Variational Inference with Inverse Autoregressive Flow.
+    [arXiv:1606.04934]
+
+    [2] Danilo Jimenez Rezende, Shakir Mohamed. Variational Inference with
+    Normalizing Flows. [arXiv:1505.05770]
+
+    [3] Mathieu Germain, Karol Gregor, Iain Murray, Hugo Larochelle. MADE: Masked
+    Autoencoder for Distribution Estimation. [arXiv:1502.03509]
+
+    """
+
+    domain = constraints.real
+    codomain = constraints.real
+    bijective = True
+    event_dim = 1
+
+    def __init__(self, hypernet, **kwargs):
+        super().__init__()
+        self.nn = hypernet
+        self.kwargs = kwargs
+
+    def condition(self, context):
```

Comment:
Consider adding a small docstring for `condition`.

Reply:
Done!